### PR TITLE
add support for ShadowMapBlock

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15289/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15289/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15289/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15289/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15289/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15289/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15289/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15289/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15289/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15289/merge/testResults/webgpuplaywright/index.html

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/shadowMapBlock.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import type { NodeMaterialBuildState } from "../../nodeMaterialBuildState";
 import { NodeMaterialBlockTargets } from "../../Enums/nodeMaterialBlockTargets";
 import type { NodeMaterialConnectionPoint } from "../../nodeMaterialBlockConnectionPoint";
 import { RegisterClass } from "../../../../Misc/typeStore";
+import { ShaderLanguage } from "core/Materials/shaderLanguage";
```
Comment (sebavan) [inline packages/dev/core/src/Materials/Node/Blocks/Fragment/shadowMapBlock.ts:7]: should it not be ../../../../ instead of core for consistency ?

**Review by sebavan — APPROVED** [review]

File: packages/dev/core/src/Materials/Node/Blocks/Fragment/shadowMapBlock.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Materials/Node/Blocks/Fragment/shadowMapBlock.ts:7]: I think they are the same now @RaananW ?

Comment (RaananW) [reply packages/dev/core/src/Materials/Node/Blocks/Fragment/shadowMapBlock.ts:7]: yes they are. Just cosmetic

**Pull request merged**